src/components/dashboard/WhoContent.native.js

```diff
@@ -0,0 +1,202 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { FlatList, PermissionsAndroid } from 'react-native'
+import { promisify } from 'es6-promisify'
+import contacts from 'react-native-contacts'
+import { map, memoize, orderBy, uniq } from 'lodash'
+import { isAndroid } from '../../lib/utils/platform'
+import { Section } from '../common'
+import InputText from '../common/form/InputText'
+import { withStyles } from '../../lib/styles'
+import normalize from '../../lib/utils/normalizeText'
+import { getDesignRelativeHeight } from '../../lib/utils/sizes'
+import userStorage from '../../lib/gundb/UserStorage'
+import FeedContactItem from './FeedContactItem'
+import ItemSeparator from './ItemSeparator'
+const Contacts = promisify(contacts.getAll)
+
+const WhoContent = ({ styles, setContact, error, text, value, next, state, showNext, setValue }) => {
+  const [contacts, setContacts] = useState([])
+  const [recentFeedItems, setRecentFeedItems] = useState([])
+  const [recentlyUsedList, setRecentlyUsedList] = useState([])
+  let initialList = useRef()
+
+  const getUserFeed = async () => {
+    const userFeed = await userStorage.getFeedPage(5, true)
+    const recent = userFeed.filter(({ type }) => type === 'send' || type === 'receive')
+    setRecentFeedItems(recent)
+  }
+
+  const showPermissionsAndroid = () => {
+    PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.READ_CONTACTS, {
+      title: 'Contacts',
+      message: 'We need access to view your contacts, so you can easily send G$ to them.',
+      buttonPositive: 'Approve',
+    })
+  }
+
+  useEffect(() => {
+    getUserFeed()
+  }, [])
+
+  useEffect(() => {
+    if (Contacts) {
+      if (isAndroid) {
+        showPermissionsAndroid()
+      }
+      Contacts().then((contacts, err) => {
+        if (err === 'denied') {
+          console.warn('permissions denied')
+        } else {
+          const sortContacts = orderBy(contacts, ['givenName'])
+          setContacts(sortContacts)
+          initialList = sortContacts
```

Comment:
https://reactjs.org/docs/hooks-reference.html#useref

useRef returns a mutable ref object whose **.current property** is initialized to the passed argument (initialValue). The returned object will persist for the full lifetime of the component.

So
1. you should access to initialList.current
2. initialList could be omited from useMemo / Callback deps

e.g.
```javascript
initialList.current = sortContacts
```